packages/app-core/src/ui/App/ViewContainer.tsx

```diff
@@ -51,8 +58,35 @@ export default observer(function ({
     scrollRef.current?.scrollIntoView?.({ block: 'center' })
   }, [])
 
+  useEffect(() => {
+    function handleSelectView(e: Event) {
+      if (e.target instanceof Element) {
+        if (ref?.current && ref.current.contains(e.target)) {
+          // @ts-ignore
+          getRoot(view).setFocusedViewId(view.id)
+        }
+      }
+    }
+
+    document.addEventListener('mousedown', handleSelectView)
+    document.addEventListener('keydown', handleSelectView)
+    return () => {
+      document.removeEventListener('mousedown', handleSelectView)
+      document.removeEventListener('keydown', handleSelectView)
+    }
+  }, [ref, view])
```

Comment:
it is likely unneeded to put ref in dependency array https://epicreact.dev/why-you-shouldnt-put-refs-in-a-dependency-array/

Reply:
I included it in the dependency array because of a lint warning: `76:6  warning  React Hook useEffect has missing dependencies: 'ref'. Either include it or remove the dependency array  react-hooks/exhaustive-deps`

Which that article does mention:
> The reason we get the warning there when we didn't get it within the component is because ESLint is pretty limited in its ability to trace what you're doing with your JavaScript. So the React plugin for ESLint can't know that cbRef is actually a ref. So just to be safe, it warns you.

> Luckily, as we just learned, including it in the dependency array doesn't make any difference anyway, so just include it!

Let me know if maybe I'm reading that wrong.